src/client/datascience/interactive-ipynb/trustCommandHandler.ts

```diff
@@ -90,42 +86,17 @@ export class TrustCommandHandler implements IExtensionSingleActivationService {
         sendTelemetryEvent(Telemetry.TrustNotebook);
     }
     private async trustNativeNotebook(model: VSCodeNotebookModel) {
-        const trustedNotebookInTrustService = createDeferred<void>();
-        const doc = this.vscNotebook.notebookDocuments.find((item) => this.fs.arePathsSame(item.uri, model.file));
-        let fileReverted = false;
-        const disposable = this.vscNotebook.onDidChangeNotebookDocument((e) => {
-            if (e.document !== doc) {
-                return;
-            }
-            trustedNotebookInTrustService.promise
-                .then(() => {
-                    // Notebook has been trusted, revert the changes in the document so that we re-load the cells.
-                    // This is a hacky solution for trusting native notebooks.
-                    if (!fileReverted) {
-                        fileReverted = true;
-                        return commands.executeCommand('workbench.action.files.revert');
-                    }
-                })
-                .catch(noop);
-            disposable.dispose();
-        });
-        this.disposables.push(disposable);
-
-        try {
-            // Update model trust
-            await model.trustNotebook();
-            // Trust the original contents & contents generated vy using the VSC Notebook model.
-            // When generating JSON from VSC, sometimes contents can get auto formatted based on user settings (hence that JSON in ipynb could be different from original contents).
-            // Thus trust both, the original content & the new content generated by VSC.
-            const originalContents = model.getOriginalContentOnDisc();
-            const contents = model.getContent();
-            await Promise.all([
-                this.trustService.trustNotebook(model.file, contents),
-                this.trustService.trustNotebook(model.file, originalContents)
-            ]);
-            sendTelemetryEvent(Telemetry.TrustNotebook);
-        } finally {
-            trustedNotebookInTrustService.resolve();
-        }
+        // Update model trust
+        await model.trustNotebook();
+        // Trust the original contents & contents generated vy using the VSC Notebook model.
+        // When generating JSON from VSC, sometimes contents can get auto formatted based on user settings (hence that JSON in ipynb could be different from original contents).
+        // Thus trust both, the original content & the new content generated by VSC.
```

Comment:
Code looks good. But I just want to be careful about this here. I understand the comment here, but technically this is trusting both the original disk content and the content as formatted by VSC. The formatting done by VSC is something that is not specifically in our control. If we trust both A.ipynb and A.ipynb as formatted by vs code are we possibly also trusting B.ipynb where B.ipynb on disk is the same as A when formatted by VS Code?

If formatting touched no real content this wouldn't be an issue but I'm not sure if we have that guarantee.

Reply:
Yes, that's correct we trust both formatted & unformatted models.
The trust state of A.ipynb & B.ipynb are not stored together, they are always different (trust state is keyed off the file Uri).